src/header-generator.js

```diff
@@ -0,0 +1,310 @@
+    }
+
+    return {
+        name: nameVersionSplit[0],
+        version: preparedVersion,
+        completeString: browserString,
+    };
+}
+
+const browserSpecificationShape = {
+    name: ow.string,
+    minVersion: ow.optional.number,
+    maxVersion: ow.optional.number,
+    httpVersion: ow.optional.string,
+};
+
+const headerGeneratorOptionsShape = {
+    browsers: ow.optional.array.ofType(ow.object.exactShape(browserSpecificationShape)),
+    operatingSystems: ow.optional.array.ofType(ow.string),
+    devices: ow.optional.array.ofType(ow.string),
+    locales: ow.optional.array.ofType(ow.string),
+    httpVersion: ow.optional.string,
+};
+
+/**
+ * @typedef BrowserSpecification
+ * @param {string} name - One of `chrome`, `firefox` and `safari`.
+ * @param {number} minVersion - Minimal version of browser used.
+ * @param {number} maxVersion - Maximal version of browser used.
+ * @param {string} httpVersion - Http version to be used to generate headers (the headers differ depending on the version).
+ *  Either 1 or 2. If none specified the httpVersion specified in `HeaderGeneratorOptions` is used.
+ */
+/**
+ * @typedef HeaderGeneratorOptions
+ * @param {Array<BrowserSpecification>} browsers - List of BrowserSpecifications to generate the headers for.
+ * @param {Array<string>} operatingSystems - List of operating systems to generate the headers for.
+ *  The options are `windows`, `macos`, `linux`, `android` and `ios`.
+ * @param {Array<string>} devices - List of devices to generate the headers for. Options are `desktop` and `mobile`.
+ * @param {Array<string>} locales - List of at most 10 languages to include in the
+ *  [Accept-Language](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language) request header
+ *  in the language format accepted by that header, for example `en`, `en-US` or `de`.
+ * @param {string} httpVersion - Http version to be used to generate headers (the headers differ depending on the version).
+ *  Can be either 1 or 2. Default value is 2.
+ */
+
+/**
+ * HeaderGenerator randomly generates realistic browser headers based on specified options.
+ */
+class HeaderGenerator {
+    /**
+     * @param {HeaderGeneratorOptions} options - default header generation options used unless overridden
+     */
+    constructor(options = {}) {
+        ow(options, 'HeaderGeneratorOptions', ow.object.exactShape(headerGeneratorOptionsShape));
+        this.defaultOptions = options;
+        this.headersOrder = JSON.parse(fs.readFileSync(headersOrderFilePath, { encoding: 'utf8' }));
+        const uniqueBrowserStrings = JSON.parse(fs.readFileSync(browserHelperFilePath, { encoding: 'utf8' }));
+        this.uniqueBrowsers = [];
+        for (const browserString of uniqueBrowserStrings) {
+            if (browserString === missingValueDatasetToken) {
+                this.uniqueBrowsers.push({
+                    name: missingValueDatasetToken,
+                });
```

Comment:
Why push this empty value into the array? This would deserve a comment at least.